cli/impl/src/main/java/io/helidon/build/cli/impl/CommonOptions.java

```diff
@@ -81,7 +84,10 @@ final class CommonOptions {
                   @KeyValue(name = "url", description = "Metadata base URL", visible = false) String metadataUrl,
                   @Flag(name = "reset", description = "Reset metadata cache", visible = false) boolean resetCache,
                   @KeyValue(name = "since", description = "Check for updates since this version",
-                          visible = false) String since) {
+                          visible = false) String since,
+                  @KeyValue(name = PROPS_FILE_OPTION_NAME, description = PROPS_FILE_OPTION_DESCRIPTION,
```

Comment:
Do we have a reason to define `--props-file` here ? The other global options are overridden here to provide a getter.

Reply:
Without `propsFile` in `CommonOptions` user will get an error `error: Unknown command option: props-file` trying to use this option.

Reply:
As for about `error: Unknown command option: props-file`:
The exception is thrown in `io.helidon.build.cli.harness.CommandParser.parseCommand(java.util.Map<java.lang.String,io.helidon.build.cli.harness.CommandParameters.ParameterInfo<?>>)` . 
The reason is that variable `paramInfo` in this method in the line  `ParameterInfo<?> paramInfo = parametersMap.get(optionName);` is equal to `null`.
It happens because `InitCommandModel` does not contains parameter `props-file`.
To add this parameter to `InitCommandModel` the code `@KeyValue(name = PROPS_FILE_OPTION_NAME, description = PROPS_FILE_OPTION_DESCRIPTION, visible = false) String propsFile` has to be added to the constructor of the class `CommonOptions`

Reply:
Why? Init command and other commands work in the same way with other params.

Reply:
A global option should be fully supported by the harness.

When you define an option in `CommonOptions` and that option is global this is basically an override. The current set of overrides seems to be a workaround for the inability to access the values of these options. I'll file an enhancement.

The real issue is that the current "global" support is only for `@Flag`, and not for `@KeyValue` and `@KeyValues`.
We need to update `GlobalOptions` and `CommandParser` to expand the notion of "global". 

- `GlobalOptions` needs to made more generic to handle `NamedOptionInfo<?>` and  `ParameterInfo<?>`
- CommandParser needs to be refactored to re-use the code that parses key value(s) needs for global options